### PR TITLE
feat: Add ModMenu support for Fabric and configuration screen for NeoForge [skip ci]

### PR DESCRIPTION
**Pull request by ShadowCat117**

### CODE REVIEW

File: fabric/build.gradle [hunk 1]
```diff
@@ -25,11 +25,19 @@ repositories {
...
+        url "https://maven.terraformersmc.com/"
+    }
 }
 
 dependencies {
     modImplementation "net.fabricmc:fabric-loader:${fabric_loader_version}"
     include(modImplementation(fabricApi.module("fabric-resource-loader-v0", fabric_version)))
+    include(modImplementation(fabricApi.module("fabric-screen-api-v1", fabric_version)))
+    include(modImplementation(fabricApi.module("fabric-key-binding-api-v1", fabric_version)))
+    include(modImplementation(fabricApi.module("fabric-lifecycle-events-v1", fabric_version)))
```
Comment (kristofbolyai) [inline fabric/build.gradle]: Are these actually needed? From what I understand, modmenu is optional. _If modmenu is present_, then these become dependencies of modmenu, not us.

Comment (magicus) [reply fabric/build.gradle]: I think you are correct. All that is needed for ModMenu is to provide a proper implementation of its `ModMenuApi` class.

File: common/src/main/java/com/wynntils/screens/settings/WynntilsBookSettingsScreen.java [hunk 2]
```diff
@@ -255,6 +259,7 @@ protected void doInit() {
 
     @Override
     public void doRender(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTick) {
+        this.renderBackground(guiGraphics, mouseX, mouseY, partialTick);
```
Comment (magicus) [inline common/src/main/java/com/wynntils/screens/settings/WynntilsBookSettingsScreen.java]: Why is this change necessary?

Comment (ShadowCat117) [reply common/src/main/java/com/wynntils/screens/settings/WynntilsBookSettingsScreen.java]: Otherwise, the modmenu background is still visible and creates some visual issues. Can't call the super render method either as that also causes issues so just need the background

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/screens/settings/WynntilsBookSettingsScreen.java]: What if we override renderBackground so it only renders a background if "McUtils.mc().level" is null?

Comment (ShadowCat117) [reply common/src/main/java/com/wynntils/screens/settings/WynntilsBookSettingsScreen.java]: https://github.com/Wynntils/Artemis/pull/2703/commits/5223418af691789fb205d36553bf12fe177f1d57 Like this?

**Review by kristofbolyai — APPROVED** [review]

File: common/src/main/java/com/wynntils/screens/settings/WynntilsBookSettingsScreen.java (hunk 2 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/screens/settings/WynntilsBookSettingsScreen.java]: Yes
